Fabulous.XamarinForms/src/Fabulous.XamarinForms.Core/ViewUpdaters.fs

```diff
@@ -28,6 +28,115 @@ module ViewUpdaters =
             if target <> null then currValue.UpdateIncremental(prevValue, target)            
         | ValueSome _, ValueNone ->
             clearValue ()
+           
+    /// Incremental list maintenance: given a collection, and a previous version of that collection, perform
+    /// a reduced number of clear/move/create/update/remove operations
+    ///
+    /// The algorithm will try in priority to update elements sharing the same instance (usually achieved with dependsOn)
+    /// or sharing the same key. All other elements will try to reuse existing previous elements where possible.
+    /// If no reuse is possible, the element will create a new control.
+    let updateCollectionGenericInternal
+            (prevCollOpt: 'T[] voption)
+            (collOpt: 'T[] voption)
+            (getKey: 'T -> string voption)
+            (canReuse: 'T -> 'T -> bool)
+            (move: int -> 'T -> unit)
+            (create: int -> 'T -> unit)
+            (update: int -> 'T -> 'T -> unit)
+            (remove: int->'T seq -> unit)
```

Comment:
> And you were right about the "replace" tests, it should be "create + remove".
> I changed them.

For current logic it looks like it is not needed, because we clear rest on create, and remove just does not take place because it is useless in such scenario. So, we make create and do not need remove because there is nothing to remove